src/si/helpers.stanza

```diff
@@ -88,6 +89,143 @@ public defn topo-net (pair:KeyValue<JITXObject,JITXObject|KeyValue>):
     (p2:JITXObject):
       topo-net(p1, p2)
 
+doc: \<DOC>
+A wrapper for `KeyValue` indicating that the given pair of
+objects should be connected by a short trace.
+<DOC>
+public deftype ShortTrace <: KeyValue
+public defn ShortTrace (kv:KeyValue) :
+  new ShortTrace :
+    defmethod key (this) : key(kv)
+    defmethod value (this) : value(kv) 
+defmethod print (o:OutputStream, st:ShortTrace) :
+  print(o, "ShortTrace(%_ => %_)" % [key(st) value(st)])
+
+doc: \<DOC>
+  This function will construct chained topology net consisting of
+  a sequence of "bus segments".
+
+  Given a sequence of key-value pairs of "bus segments", this function
+  will connect them, generate topology segments, and constrain the trace
+  length as necessary.
+
+  A "bus segment" is either:
+  * A tuple of single-pin or array ports
+  * A single-pin or array port
+  * An Instantiable
+  * A "ShortTrace"
+
+  The left- and right-most components of the chain must be concrete --
+  either a pin or a tuple of pins. If given an Instantiable
+  for one segment of the chain, this function will create as many concrete
+  instances as necessary.
+
+  We do not currently accept concrete instances, just pins or Instantiables.
+
+  We assume that any compatible Instantiable has pins `p[1]` and `p[2]`.
+
+  We assume that all concrete segments are the same size and shape:
+  we can only compute `topo-bus([a, b, c], [d, e, f])` when port `a`
+  matches port `d`, and so on.
+
+  Returns a tuple of all newly-instantiated objects.
+
+  @snippet
+
+  ```stanza
+  port mii : rmii-bus
+  require sw-bus : rmii-bus from mcu
+  val series-R-type = create-resistor(resistance = 51.0, precision = (1 %), case = ["0603"])
+  val rx-bus = [sw-bus.rxd, sw-bus.crs-dv, sw-bus.ref-clk]
+  val rx-out = [mii.rxd, mii.crs-dv, mii.ref-clk]
+  topo-bus(ShortTrace(rx-bus => series-R-type) => rx-out)
+  ```
+
+  @snip-note 3 `create-resistor` returns an `Instantiable`
+  @snip-note 4 `sw-bus.rxd` and `mii.rxd` are both arrays of two pins. `topo-bus` will return a tuple of four new resistors that connect the two buses.
```

Comment:
`both arrays` maybe we say `both PortArray instances of two pins`. I want to avoid confusion between `array` and `tuple` for new users.